Add pipewire support with console, qt and gtk GUI.

Pri-yan-shu opened the pull request:
Adds pipewire support, taking inspiration from jack architecture files and faust2jack files.

sletz commented:
@Pri-yan-shu AFAICS this PR seems LLM generated, without deep understanding of the code. If you can correct it and prove that is actually works, then fine, otherwise I'll close it.

Pri-yan-shu commented:
@sletz I have started from jack architecture files and faust2 scripts as base and modified the jack specific api with pipewire. All 3 scripts work and generate perfectly running programs with working midi support. If you want I can send the video demonstration over faust discord.

dromer commented:
Not super important, but shouldn't the dates in the file headers all be 2025? Since these are new files and should reflect their current status.

Pri-yan-shu commented:
@sletz Fair enough. I kept the `pw_thread_loop *loop` for `pw_thread_loop_destroy(loop)` in the destructor, but all pipewire specific code has been written by looking at pipewire's example programs, specificly [https://gitlab.freedesktop.org/pipewire/pipewire/-/blob/master/src/examples/midi-src.c](url) and [https://gitlab.freedesktop.org/pipewire/pipewire/-/blob/master/src/examples/audio-dsp-filter.c](url).

Pri-yan-shu commented:
@dromer I began with jack architecture files as base and didn't knew much about writing licence info so I kept the dates as is assuming maintainers would take care of that. I would update the dates of the files.

Pri-yan-shu commented:
@sletz I have cleaned up the code as much as I could, and fixed the licence dates. I replaced unnecessary state with requesting variables on the fly. I need `pw_thread_loop *loop` for locking and destruction (no means to get `pw_thread_loop` from `pw_filter`). Pipewire doesn't take the ownership of `filter_events` and `filter_listener` so I need to keep them as class members as well. I have tested midi and polyphonic support working fine. If you find any issues with the code point them out, I'll fix them.

sletz commented:
Thanks. 

- since I'm not a PW user at all, I would love you to properly comment the PW related code (threading, filter...) everywhere needed
-  ans also could you format the code with our naming convention, like using `fLoop` instead of `loop` (fields like `fXXX`...)

Pri-yan-shu commented:
Done

sletz commented:
Thanks. 

Can you check why this happens  ?: 

```
faust2pwconsole organ.dsp
/usr/local/bin/faust2pwconsole: line 111: [: =: unary operator expected

```

Pri-yan-shu commented:
The line is `if [ $POLY = "POLY2" ]; then`. If POLY is not assigned any value beforehand the line evaluates to `if [ = "POLY2" ]`, hence the error. I was getting same error on upstream faust2jackconsole so I assumed it to be fine. I have fixed it now, assigning `POLY="POLY"` at beginning like other scripts.

sletz commented:
Thanks a lot. 

Merged and Faust version raised to 2.80.3.
